### PR TITLE
Persistence layer support for multiple facilities 

### PR DESCRIPTION
**Pull request by benwarfield-usds**
Added a database object for facilities, and migrations such that all existing organizations have exactly one facility, then updated API handlers such that attempts to modify the fields of an organization
that now reside on its (single) facility will instead be performed on the facility object.

Also did a minor refactor to eliminate some boilerplate about street addresses, and updated many, many tests.

Gotchas:
* the database migration to populate the `facility` table requires the `pgcrypto` module to be enabled in postgresql. It will be skipped if run on an empty database, but we should enable that module for deployed environments
  - [x] run `create extension if not exists pgcrypto;` on cloud.gov database
  - [x] run `create extension if not exists pgcrypto;` on demo database
  - [x] run `create extension if not exists pgcrypto;` on staging database
  - [ ] run `create extension if not exists pgcrypto;` on prod database
  - [x] figure out how to let people run `create extension if not exists pgcrypto;` on their local dev databases without nuking.
    ```sh
    docker-compose exec -u postgres db psql -h localhost -p ${SR_DB_PORT:-5432} -c 'create extension if not exists pgcrypto' simple_report
    ```
* certain facility information is currently hard-coded in our export because it isn't in the database. This change adds support for pulling it from the database, but not for inserting it—we should probably just restore the hard-coded values until we actually support editing them through the UI.

### CODE REVIEW

File: backend/src/main/java/gov/cdc/usds/simplereport/api/model/TestEventExport.java [hunk 1]
```diff
@@ -304,32 +288,32 @@ public String getOrderingFacilityCounty() {
 
 	@JsonProperty("Ordering_facility_name")
 	public String getOrderingFacilityName() {
-		return org.getFacilityName();
+		return facility.getFacilityName();
```
Comment (timothybest-usds) [inline backend/src/main/java/gov/cdc/usds/simplereport/api/model/TestEventExport.java:291]: To maintain our current format (`"${organization.name}, ${facility.name}"`) this needs to include the organization name

Comment (timothybest-usds) [reply backend/src/main/java/gov/cdc/usds/simplereport/api/model/TestEventExport.java:291]: I am not sure on this I think the the Facility name the is associated with the CLIA number might include what looks like the organization name 🤔

Comment (timothybest-usds) [reply backend/src/main/java/gov/cdc/usds/simplereport/api/model/TestEventExport.java:291]: Confirmed with @aliciabeckett-gov this should remain as return `facility.getFacilityName();` but our current export needs to be updated

**Review by timothybest-usds — APPROVED** [review]

File: backend/src/main/java/gov/cdc/usds/simplereport/api/model/ApiFacility.java [hunk 2]
```diff
@@ -16,33 +17,33 @@ public String getName() {
 
 	public String getCliaNumber() {
 		return org.getCliaNumber();
-  }
+    }
   
-  public String getStreet() {
-		return "2797 N Cerrada de Beto";
+    public String getStreet() {
+		return org.getAddress().getStreetOne();
```
Comment (nickrobison-usds) [inline backend/src/main/java/gov/cdc/usds/simplereport/api/model/ApiFacility.java:23]: Shouldn't we be pulling the address info from the facility itself?

File: backend/src/main/java/gov/cdc/usds/simplereport/api/organization/OrganizationDataResolver.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+@Service
+public class OrganizationDataResolver implements GraphQLResolver<ApiOrganization> {
+
+	private static final Logger LOG = LoggerFactory.getLogger(OrganizationDataResolver.class);
+
+	@Autowired
+	private OrganizationService _orgService;
+
+	public ApiFacility getTestingFacility(ApiOrganization o) {
+		LOG.error("Asked me for a facility for {}", o);
```
Comment (nickrobison-usds) [inline backend/src/main/java/gov/cdc/usds/simplereport/api/organization/OrganizationDataResolver.java:28]: Why are we logging these as errors?

**Review by nickrobison-usds — CHANGES_REQUESTED** [review]
I think this looks pretty good, nice work!

A couple of small questions in the review.

One point of clarification, why do we need to add the `pgcyrpto` extension? Not sure we have permissions to do that in Azure, what functionality is that actually providing?

File: backend/src/main/java/gov/cdc/usds/simplereport/db/model/Organization.java [hunk 4]
```diff
@@ -1,132 +1,39 @@
...
+		return organizationName;
 	}
 
-	public void setFacilityName(String facilityName) {
-		this.facilityName = facilityName;
-	}
-
-	public String getFacilityName() {
-		return facilityName;
+	public void setOrganizatioName(String newName) {
```
Comment (timothybest-usds) [inline backend/src/main/java/gov/cdc/usds/simplereport/db/model/Organization.java:32]: nitpick `setOrganizatioName ` -> `setOrganizationName `

Comment (timothybest-usds) [reply backend/src/main/java/gov/cdc/usds/simplereport/db/model/Organization.java:32]: missing the n at the end of Organization

File: backend/src/main/java/gov/cdc/usds/simplereport/api/model/ApiFacility.java (hunk 2 above)
Comment (benwarfield-usds) [reply backend/src/main/java/gov/cdc/usds/simplereport/api/model/ApiFacility.java:23]: We are, it's just still named `org` because I wanted to keep the diff small and knew I was going to be redoing a bunch of this class later in the day anyway. 😸

File: backend/src/main/java/gov/cdc/usds/simplereport/api/organization/OrganizationDataResolver.java (hunk 3 above)
Comment (benwarfield-usds) [reply backend/src/main/java/gov/cdc/usds/simplereport/api/organization/OrganizationDataResolver.java:28]: Because we forgot to take out the debug code from when we were making sure that we understood how `GraphQLResolver<?>` beans worked. 😳

File: backend/src/main/java/gov/cdc/usds/simplereport/db/model/Organization.java (hunk 4 above)
Comment (benwarfield-usds) [reply backend/src/main/java/gov/cdc/usds/simplereport/db/model/Organization.java:32]: Thanks for the second comment, I could still barely see the typo after you pointed it out. &lt;sigh&gt;

**Comment by benwarfield-usds** [comment]
> One point of clarification, why do we need to add the `pgcyrpto` extension? Not sure we have permissions to do that in Azure, what functionality is that actually providing, generating the UUIDs?

Yes, that's exactly what it's doing—I just double checked and we do have it in the managed instance, but thanks for the flag! That would have sucked.

**Review by nickrobison-usds — APPROVED** [review]

File: backend/src/main/java/gov/cdc/usds/simplereport/api/model/TestEventExport.java [hunk 5]
```diff
@@ -304,32 +288,32 @@ public String getOrderingFacilityCounty() {
...
 	}
 
 	@JsonProperty("Ordering_facility_phone_number")
 	public String getOrderingFacilityPhoneNumber() {
-		return "5202475313";
+		return facility.getTelephone();
```
Comment (benwarfield-usds) [inline backend/src/main/java/gov/cdc/usds/simplereport/api/model/TestEventExport.java:296]: @timothybest-usds should I revert the changes to this file so we can just merge this without breaking the export?

Comment (timothybest-usds) [reply backend/src/main/java/gov/cdc/usds/simplereport/api/model/TestEventExport.java:296]: no worries about the export. next one isn't until 9 and I can add them manually, update the facility, or preferred merge in the edit facility information PR (should be rebased on this branch and working by end of day)

Comment (benwarfield-usds) [reply backend/src/main/java/gov/cdc/usds/simplereport/api/model/TestEventExport.java:296]: We should coordinate offline on that last bit, I had started some schema iteration that might profitably mix in here—but otherwise cool, I shall fire away!
